spring-batch-integration/src/main/java/org/springframework/batch/integration/config/annotation/BatchIntegrationConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.batch.integration.config.annotation;
+
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.batch.integration.chunk.RemoteChunkingMasterStepBuilderFactory;
+import org.springframework.batch.integration.chunk.RemoteChunkingWorkerFlowBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.transaction.PlatformTransactionManager;
+
+/**
+ * Base configuration class for Spring Batch Integration factory beans.
+ *
+ * @since 4.1
+ * @author Mahmoud Ben Hassine
+ */
+@Configuration
+public class BatchIntegrationConfiguration {
+
+	private JobRepository jobRepository;
+
+	private PlatformTransactionManager transactionManager;
+
+	private GenericApplicationContext applicationContext;
+
+	@Autowired
+	public BatchIntegrationConfiguration(
+			JobRepository jobRepository,
+			PlatformTransactionManager transactionManager,
+			GenericApplicationContext applicationContext) {
```

Comment:
See previous comment about `BeanDefinitionRegistry`.

Reply:
done

Reply:
The code here says otherwise...

Reply:
fixed, my mistake, sorry for that.